### PR TITLE
[Accessibility] Announce notice messages after page loads

### LINKED ISSUE #43614: Accessibility: Add to cart on single product not announced to screen readers
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When adding to the cart on a product single, the page reloads, and a banner notice is visible:
Screenshot 2024-01-14 at 5 07 44 PM

This banner is not announced to screen reader users, and unless they explore the page, they will not know if their add-to-cart was successful.

Because this message appears above the H1, if they are using heading shortcuts to navigate the page, they are likely to miss the banner.

### Expected behavior

When a product is added to the cart, there should be an auditory message for screen reader users to know that the add-to-cart action was successful.

You can either do this by:
A) When the page reloads, set keyboard focus on the status message so it is read out. OR
B) Add an [aria-live](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) region on the status message and trigger it after the page loads.

### Actual behavior

Screen reader users don't easily know when a product is added to the cart.

### Steps to reproduce

1. Turn on a screen reader such as VoiceOver or NVDA.
2. Go to the single product page.
3. Add a product to your cart.
4. Listen as the page reloads.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### LINKED ISSUE #51929: Accessibility: Status message not announced by screen readers [Product Single (Grouped)]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The error message if the Add to Cart button is selected without choosing products is visually displayed, but screen readers do not announce it (though it has a role of alert) because the page refreshes and screen readers instead read the page title/top of the page.

Image

### Component or Page: 

Page: Product Single (Grouped)  

### Name affected element:

Error message if Add to Cart button is selected without choosing products

### Relevant code snippets:

`<div class=""wc-block-components-notice-banner is-error"" role=""alert"">
	<svg xmlns=""http://www.w3.org/2000/svg"" viewBox=""0 0 24 24"" width=""24"" height=""24"" aria-hidden=""true"" focusable=""false"">
		<path d=""M12 3.2c-4.8 0-8.8 3.9-8.8 8.8 0 4.8 3.9 8.8 8.8 8.8 4.8 0 8.8-3.9 8.8-8.8 0-4.8-4-8.8-8.8-8.8zm0 16c-4 0-7.2-3.3-7.2-7.2C4.8 8 8 4.8 12 4.8s7.2 3.3 7.2 7.2c0 4-3.2 7.2-7.2 7.2zM11 17h2v-6h-2v6zm0-8h2V7h-2v2z""></path>
	</svg>
	<div class=""wc-block-components-notice-banner__content"">
		Please choose a product to add to your cart…	</div>
</div>`

### Recommended fix:

Possible options to remediate this:

A) Don't refresh the page on submit of the ""Add to Cart"" button and instead use JavaScript to triggert the action/show the error.

B) After the page is refreshed, use JavaScript to set keyboard focus on the error message so screen readers read it rather than the content at the top of the page.

### LINKED ISSUE #51931: Accessibility: Status message not announced by screen readers [Product Single (All Types)]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The add to cart success message is visually displayed, but screen readers do not announce it (though it has a role of alert) because the page refreshes and screen readers instead read the page title/top of the page.

Image

### Component or Page: 

Page: Product Single (All types)  

### Name affected element:

Success message for adding product to cart

### Relevant code snippets:

```
<div class=""woocommerce-notices-wrapper"">
	<div class=""wc-block-components-notice-banner is-success"" role=""alert"">
		<svg xmlns=""http://www.w3.org/2000/svg"" viewBox=""0 0 24 24"" width=""24"" height=""24"" aria-hidden=""true"" focusable=""false"">
			<path d=""M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z""></path>
		</svg>
		<div class=""wc-block-components-notice-banner__content"">
			<a href=""https://wooprd.wpenginepowered.com/cart/"" tabindex=""1"" class=""button wc-forward wp-element-button"">View cart</a> “Intelligent Paper Knife” and “Durable Granite Knife” have been added to your cart.		</div>
	</div>
</div>
```

### Recommended fix:

Possible options to remediate this:

A) Don't refresh the page on submit of the ""Add to Cart"" button and instead use JavaScript to triggert the action/show the error.

B) After the page is refreshed, use JavaScript to set keyboard focus on the success message so screen readers read it rather than the content at the top of the page.

This is connected to https://github.com/woocommerce/woocommerce/issues/51929

### LINKED ISSUE #52142: Accessibility: Content reading order does not follow meaningful order [Add to Cart]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

When the setting to redirect to the cart page after add to cart is turned on, there is a status message that appears at the top of the cart page after each add to cart action. When listening to this status message with a screen reader, the content is heard out of order or read in a way that is confusing or lacks meaning.   

Illogical reading order is being observed:
1.Continue Shopping button >> 2. Status message “Aerodynamic Steel Table” has been added to your cart.

Image

### Component or Page: 

Add to cart redirect setting toggled on

### Name affected element:

Status message “Product Name” has been added to your cart and Continue Shopping button on cart page after adding product to cart

### Relevant code snippets:

`<div class=""wc-block-components-notice-banner__content"">
			<a href=""https://woo.equalizedigital.com/?product=aerodynamic-steel-table"" tabindex=""1"" class=""button wc-forward wp-element-button"">Continue shopping</a> “Aerodynamic Steel Table” has been added to your cart.		</div>`

### Recommended fix:

After redirecting to the cart page, ensure that the first thing screen readers read out is the success message.

Change the DOM order so the continue shopping button ocurrs after the status message (as it visually appears).

### PR DESCRIPTION
**Pull request by Manussakis**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

1. This PR introduces a JS code that looks for any notice element with the `role="alert"` attribute on the page and focuses it once the page loads.
2. The markup of the success message of the add-to-cart action has been changed. There were two accessibility issues with the old markup:
    1. The visual order of the elements didn't match the DOM order, so the focus changing didn't follow a logical order.
    2. Once the notice message was focused, the link's content was announced before the message itself.

#### Before
The link comes before the message.
image
image
image

#### After
The link comes after the message.
image
notice-add-to-cart
notice-stock

The change in the markup doesn't entail a significant change in the visual appearance. In fact, two other notice messages had already the link coming after the text. They are:
- My Account -> Downloads
notice-no-downloads
- My Account -> Order
notice-no-orders

Closes #43614 .
Closes #51931 .
Closes #51929 .
Closes #52142 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Testing the add-to-cart notice
1. Install the Twenty Twenty-Four theme.
2. Go to a Product page.
3. Add the product to the cart.
4. Verify the error message gets the focus once the page loads.

#### Testing the stock notice
1. Edit the stock quantity of a product to have only one unit available.
2. Go to the product page on the frontend.
3. Add two products to the cart.
4. Verify the error message gets the focus once the page loads.

#### Testing the Address form notices
1. On the frontend, go to My Account -> Addresses -> Edit address.
2. Leave a required field empty and submit the form.
3. Verify the error message gets the focus once the page loads.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/wc-cart-functions.php
Comment (Manussakis) [inline plugins/woocommerce/includes/wc-cart-functions.php]: Add a wrapper to the message and the link and swap their position in the DOM.

**Comment by Manussakis** [comment]
@dkotter This PR is ready for code review.

**Review by dkotter — APPROVED** [review]
@ankitguptaindia @qasumitbagthariya This is ready for QA

**Comment by dkotter** [comment]
@Manussakis One thing I just noticed, looks like this PR is causing one of the unit tests to fail. That will need to be fixed up before this can be merged in: https://github.com/woocommerce/woocommerce/actions/runs/10133808611/job/28019382690?pr=50061#step:7:121

**Comment by Manussakis** [comment]
Thanks for spotting that, @dkotter . Somehow I didn't associate it with my changes.
@ankitguptaindia @qasumitbagthariya I have fixed the tests and this task is ready for QA.

**Comment by nielslange** [comment]
I'll remove myself from the review, given that this PR affects the product page, while I'm maintaining the Cart and Checkout blocks.

**Comment by ankitguptaindia** [comment]
Thanks for the PR @Manussakis this PR works as explained but I have some points to confirm:

1. When a product is added to the cart, a success message appears and announces "YourProduct has been added to your cart." However, this message is announced twice. Is this related to this PR, or can we fix this here?
2. When the message announces "YourProduct has been added to your cart," it also announces "1 more item," but there is no other item in the cart or on the page.

**Please watch this video explanation for more details:**  



https://github.com/user-attachments/assets/50eb4555-9635-4587-8ff7-034603ff57ee






QA Status-  WIP, will add more points tomorrow.

**Comment by Manussakis** [comment]
@ankitguptaindia Thanks for testing this PR!
> 1. When a product is added to the cart, a success message appears and announces "YourProduct has been added to your cart." However, this message is announced twice. Is this related to this PR, or can we fix this here?
> 2. When the message announces "YourProduct has been added to your cart," it also announces "1 more item," but there is no other item in the cart or on the page.

This is how VoiceOver works in this context. The message it announces when the alert element is initially focused is like a summary:
`'"Belt" has been added to your cart, and 1 more item, alert, main'`
Let's break it down:
- `"Belt" has been added to your cart` – This is the first element in the alert.
- `and 1 more item` – Since it's a summary, VoiceOver doesn't announce all the elements now, so it only announces that in addition to the message before there is one more item inside the alert element, which is the "View cart" link on the right side of the alert.
- `alert,` – because the element has the `role="alert"` attribute.
- `main` – because the element is inside the `<main>` HTML tag.

Next, VoiceOver starts reading each one of the elements inside the alert. That's why the message is announced again. Notice that after announcing the message it announces `visited, link, View cart`, which is the "View cart" link on the right side.

Other screen readers may address alert messages differently, but this is something that users should update via screen reader preferences.

**Comment by ankitguptaindia** [comment]
QA Report: Needs feedback ⚠️ 
---

@Manussakis Thanks for the detailed explanation https://github.com/woocommerce/woocommerce/pull/50061#issuecomment-2259089291. 


-----
Regarding the PR workflow: 

Part-1:  `Verify that after the page loads, the focus goes to the success message at the top of the page` 

It works as intended.  ✅ 

Demo: 

<Details>


https://github.com/user-attachments/assets/05609b26-a31f-4c7a-a87c-ef78cab93359

</Details>

-----------

Part -2: `Verify the error message gets the focus once the page loads.`. ⚠️ 

It works fine when editing the billing address but if editing the shipping address, sometimes it doesn't announce the alert message: 

<Details>

https://github.com/user-attachments/assets/45d27073-5e60-437e-a975-170637767c3b

</Details>

When trying with a new user in different tab, it doesn't announce the alert message: 


<Details>

https://github.com/user-attachments/assets/efe64546-843f-4d07-a94c-bc2c4221e6ba

</Details>

With Screen Reader Chrome extension, it works fine for same page: 

<Details>

https://github.com/user-attachments/assets/b36f5c76-373e-4f4a-8218-1b91c680715d

</Details>

**Comment by Manussakis** [comment]
@ankitguptaindia Thanks for the feedback!
Part-2:
Good catch! I have updated the code.
Since Local returns AJAX requests almost instantly, could you also test this PR with a slower connection to simulate a more realistic scenario?

@dkotter Please review this change: [Cobine delay with queue method](https://github.com/woocommerce/woocommerce/pull/50061/commits/72d8efb993637338b46df1046e29211d96929f49)

**Review by dkotter — APPROVED** [review]

**Comment by ankitguptaindia** [comment]
**QA/Test Report-** ✅ 
------------

**Testing Environment -** 

<Details>

* WordPress: 6.6.1
* Theme: 
**Storefront Version: 4.6.0  
**Twenty Twenty-Four Version: 1.2
* WooCommerce - Development version with this fix-specific branch.  
* PHP: 8.3
* Web Server: Nginx
* Browser: Chrome - Version 127
* OS: macOS Sonoma 14.5
* Git Branch: fix/43614


</Details>


**Test Results -** 

Tested with Safari, Firefox and Chrome, Changes work as expected. Tested via macOS Voice Over and Screen Reader Chrome extension. Tested with Storefront theme and Twenty-Twenty-Four theme
 

**Functional Demo /  Screencast -**

**With macOS Voice Over:**


https://github.com/user-attachments/assets/85846ffc-0f8e-46a9-8440-8ab2278cb766

**Comment by ankitguptaindia** [comment]
@Manussakis PR changes LGTM. Please Un-draft this PR for the next step.

**Comment by Manussakis** [comment]
@vikrampm1 This task is ready to be reviewed by the appropriate Woo team.

**Comment by jeffpaul** [comment]
@woocommerce/woo-fse this PR relates to an issue labeled for your team and the PR is now ready for your review

**Comment by imanish003** [comment]
Thanks, @jeffpaul! As I am assigned to this PR, I can review it from the Woo FSE perspective. However, I believe it will require a review from @woocommerce/Rubik as well, as it touches files around the Cart and Checkout code.

**Comment by alexflorisca** [comment]
> This PR introduces a new CSS class named woocommerce-notice-content used on the element that wraps the message and the link on the add-to-cart success message.

@Manussakis Why do we need the wrapper here? 

> Themes that override the WooCommerce default styles must update their CSS files to include styles for the woocommerce-notice-content CSS class.

This is quite worrying to me as it seems this has the potential for breaking a lot of themes. Could we find a way to avoid this?

**Comment by Manussakis** [comment]
@alexflorisca, thanks for the feedback!

> Why do we need the wrapper here?

The wrapper holds the flex behavior that allows the order of the elements to be swapped while preserving the current visual alignment.

> This is quite worrying to me as it seems this has the potential for breaking a lot of themes. Could we find a way to avoid this?

I will look into it further.

**Comment by Manussakis** [comment]
@alexflorisca I updated the PR description to explain the new changes after your feedback.

@dkotter Please re-review this PR.

**Review by dkotter — APPROVED** [review]

**Comment by jeffpaul** [comment]
@imanish003 @alexflorisca this is back to you for review, thanks!

**Comment by alexflorisca** [comment]
I'm happy with the code changes and the approach here now, thank you for making those changes! But I'm off until next week and won't have time to properly test. So if @dkotter has tested it then we can merge, I don't see any problems with Cart & Checkout here.

**Comment by dkotter** [comment]
@ankitguptaindia I know you tested this before but there's been some changes since (which you can read through the thread on this issue for full details). Can you test this one again before we proceed with merging?

**Comment by ankitguptaindia** [comment]
Re-tested PR after new changes in Safari, Chrome and Firefox with macOS 14.5VoiceOver.

https://github.com/user-attachments/assets/5e9a7d86-8f64-42fc-b055-3aed117d9544



**Next Step-** PR is ready for merge. 🚀

**Comment by Manussakis** [comment]
@vikrampm1 This PR is ready for the next step.

**Comment by vikrampm1** [comment]
@imanish003 / @woocommerce/woo-fse this PR is ready for your review!

File: plugins/woocommerce/client/legacy/js/frontend/woocommerce.js
Comment (imanish003) [inline plugins/woocommerce/client/legacy/js/frontend/woocommerce.js]: Can we please avoid using JQuery in this PR? Instead, can we use Vanilla JavaScript with `IntersectionObserver` or `MutationObserver` to achieve the same functionality? Internally, there was a discussion (pcShBQ-1S4-p2) around it with title "Please don’t write any more JQuery in WooCommerce" by @samueljseay. Based on this @ObliviousHarmony also created issue #48589. As we are making efforts to get rid of JQuery, it would be great if we can avoid using it in this PR. Otherwise, it will introduce another peace of code that will eventually need to be migrated. 

What do you think?

**Review by imanish003 — CHANGES_REQUESTED** [review]
Hey @Manussakis,

Thanks for working on improving the accessibility of the notice messages 🚀 I have left one comment on the PR.

File: plugins/woocommerce/client/legacy/js/frontend/woocommerce.js
Comment (senadir) [reply plugins/woocommerce/client/legacy/js/frontend/woocommerce.js]: I'd advice against using Vanilla here, at least, not using `IntersectionObserver` and `MutationObserver`. It make sense to stop using jQuery but this is already an existing script that does exactly that. I understand @samueljseay advised against that but unless we have some good effort to get rid of it, we would be just fragmenting files.

If we're to go with vanilla, I'd be wary of observers, they're a bit recentish and we don't ship any polyfills for them, we hade some reports in Checkout block of a Checkout not loading due to the browser (usally an old phone or an in-app android browser like facebook) not supporting that.

Comment (Manussakis) [reply plugins/woocommerce/client/legacy/js/frontend/woocommerce.js]: Thanks @imanish003 @senadir !
I will refactor the code to use Vanilla JS without using observers.

**Comment by Manussakis** [comment]
@imanish003 @dkotter I have updated the code to use vanilla JS, please review it: [Refactor to vanilla JS code that focus on notice message](https://github.com/woocommerce/woocommerce/pull/50061/commits/186d2a20fa9919e5e3a4a71b7268adab869fd565)

@ankitguptaindia This PR will need another round of QA once the code is approved.

**Review by dkotter — APPROVED** [review]

**Comment by Manussakis** [comment]
@ankitguptaindia Please test this PR again 🙏

**Comment by qasumitbagthariya** [comment]
### QA Update ✅ 
-------------

I have checked this PR in the `Manussakis:fix/43614` branch and the changes are working as expected.

**Testing the add-to-cart notice** ✅

https://github.com/user-attachments/assets/e694b8b5-ee18-4d10-8d3e-df7eef0a863d

**Testing the stock notice** ✅

https://github.com/user-attachments/assets/612dc392-04c9-4493-980f-bfe1e15416cc

**Testing the Address form notices** ✅

https://github.com/user-attachments/assets/87079187-237d-4676-b8ae-150a82e2fcf2



**Testing Environment** 

<Details>

- WordPress: 6.6
- Theme: Twenty Twenty-Four 1.2
- WooCommerce - WooCommerce - Dev version with this fix specific branch.
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS Ventura 13.3 
- Git Branch: Manussakis:fix/43614

</Details>


**Status update:**
@vikrampm1  This PR is ready for the next step.

**Comment by vikrampm1** [comment]
@imanish003 @senadir this is back for your re-review!

**Review by imanish003 — APPROVED** [review]
@Manussakis Changes looks good to me. I have tested the changes locally and they are working as expected. Thanks for updating the code to use Vanilla JavaScript instead of JQuery 🙌🏻

**Comment by vikrampm1** [comment]
@imanish003 this is ready to be merged by you / Woo now. Thanks!

**Comment by ktmn** [comment]
> This PR introduces a JS code that looks for any notice element with the role="alert" attribute on the page and focuses it once the page loads.

https://github.com/woocommerce/woocommerce/issues/43614:

> You can either do this by:
> A) When the page reloads, set keyboard focus on the status message so it is read out. OR
> B) Add an [aria-live](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) region on the status message and trigger it after the page loads.

@Manussakis Is option "A" ideal? 

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role:

> The element with the alert role does not have to be able to receive focus, as screen readers (speech or braille) will automatically announce the updated content regardless of where keyboard focus when the role is added.

> Setting `role="alert"` is equivalent to setting `aria-live="assertive"` and `aria-atomic="true"`. As they don't receive focus, focus does not need to be managed and no user interaction should be required.

---

For me, the focus adds a thick black border to the alert (browser default styles):

```css
:focus-visible {
    outline: -webkit-focus-ring-color auto 1px;
}
```

I'm wondering if hurts accessibility if I were to to disable it with:

```css
.wc-block-components-notice-banner:focus-visible {
    outline: none;
}
```
as it's not actually a focusable element. For example when going to **My account -> Downloads** and there are no downloads, the message that says _"No downloads available yet."_ has a thick outline which looks like the message is styled poorly.

**Comment by Manussakis** [comment]
From #43614 

> You can either do this by:
A) When the page reloads, set keyboard focus on the status message so it is read out. OR
B) Add an [aria-live](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) region on the status message and trigger it after the page loads.

> @Manussakis Is option "A" ideal?

@ktmn Thanks for raising this question.

Live regions tend to work well across different browser/screen reader combinations, but the ideal use of it is when their content is updated dynamically via JavaScript. However, most pages in Woo come with the live region content populated on the initial markup, which doesn't trigger its announcement by screen readers.

Please, check the "Live regions that are already "populated" don't work" section of this article: https://tetralogical.com/blog/2024/05/01/why-are-my-live-regions-not-working/

Considering this scenario, focusing on the live region once the page loads becomes the better option for fixing this issue, in my opinion.

I agree with your point regarding the thick black border added to the alert when it gets the focus and would add one more selector to your suggestion:
```scss
.wc-block-components-notice-banner {
	&:focus,
	&:focus-visible {
		outline: none;
	}
}
```

**Comment by Manussakis** [comment]
@ktmn I have created the https://github.com/woocommerce/woocommerce/issues/51003 issue to tackle the focus style of the notice message.

**Comment by senadir** [comment]
Following up here to say that this PR introduced a small regression in One Page Checkout. When landing on a product page with Checkout in it, focus will jump from top of page to Checkout (which sits low in the page).

Maybe we can add some exclude classes to notices (dont-autofocus) and One Page Checkout can hook into Checkout and add those classes.

**Comment by jeffpaul** [comment]
@Manussakis @vikrampm1 can one of you please open an issue to document @senadir's issue above and drop into our a11y project board to work on resolving?

**Comment by Manussakis** [comment]
@jeffpaul I will create the issue once I reproduce it locally so I can fill the issue form out with all the bug details.
I plan to do it today.

**Comment by Manussakis** [comment]
> Following up here to say that this PR introduced a small regression in One Page Checkout. When landing on a product page with Checkout in it, focus will jump from top of page to Checkout (which sits low in the page).
> 
> Maybe we can add some exclude classes to notices (dont-autofocus) and One Page Checkout can hook into Checkout and add those classes.

@senadir Would you mind giving me more context on this issue?
I activated the One Page Checkout from the Product data panel:
image

Then I went to the product page, but the only notice message available low on the page is the Coupon one, which has the `role="status"` attribute, therefore, the focus is not jumping there since the script looks only for notice messages [with the `role="alert"` attribute](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/legacy/js/frontend/woocommerce.js#L161).

image

I have tested with both Classic and Block themes, but I was not able to reproduce it yet.

Could you please help me with that?

**Comment by vikrampm1** [comment]
@senadir can you help review the comment above?

**Comment by senadir** [comment]
Hey! I do recall it jumping to notice, though maybe it was the account creation one. I can look again today.

**Comment by vikrampm1** [comment]
@senadir following up on the above if you have any additional comments.

**Comment by vikrampm1** [comment]
@senadir #51003 and #51914 are still blocked and awaiting your response on the above if you can take a look.

**Comment by senadir** [comment]
I can no longer reproduce the issue, I'm not sure what changed, but we can consider those unblocked for now

**Pull request merged**
